angular-client/src/components/toast-button/toast-button.component.ts

```diff
@@ -0,0 +1,34 @@
+import { Component, Input } from '@angular/core';
+import { MessageService } from 'primeng/api';
+
+@Component({
+  selector: 'toast-button',
+  templateUrl: './toast-button.component.html',
+  styleUrls: ['./toast-button.component.css']
+})
+export class ToastButtonComponent {
+  @Input() label!: string;
+  @Input() onClick!: () => void;
+  @Input() additionalStyles?: string;
+  style!: string;
+
+  ngOnInit(): void {
+    this.style = 'width: 140px; height: 45px; ';
+
+    if (this.additionalStyles) {
+      this.style += this.additionalStyles;
+    }
+  }
+
+  constructor(private messageService: MessageService) {}
+
+  showSuccess() {
+    setTimeout(() => {
```

Comment:
you need to call onClick in this method so that it only displays the success toast if the onClick is successful